daemon/cmd/daemon_main.go

```diff
@@ -1840,6 +1725,134 @@ func initKubeProxyReplacementOptions() {
 				"will be selected from all network namespaces on the host.")
 		}
 	}
+
+	if option.Config.EnableNodePort {
+		if option.Config.Tunnel != option.TunnelDisabled &&
+			option.Config.NodePortMode != option.NodePortModeSNAT {
+
+			log.Warnf("Disabling NodePort's %q mode feature due to tunneling mode being enabled",
+				option.Config.NodePortMode)
+			option.Config.NodePortMode = option.NodePortModeSNAT
+		}
+
+		if option.Config.NodePortAcceleration != option.NodePortAccelerationDisabled {
+			if option.Config.Tunnel != option.TunnelDisabled {
+				log.Fatalf("Cannot use NodePort acceleration with tunneling. Either run cilium-agent with --%s=%s or --%s=%s",
+					option.NodePortAcceleration, option.NodePortAccelerationDisabled, option.TunnelName, option.TunnelDisabled)
+			}
+		}
+	}
+
+	return
+}
+
+// detectDevicesForNodePortAndHostFirewall tries to detect bpf_host devices
+// (if needed).
+func detectDevicesForNodePortAndHostFirewall(strict bool) {
```

Comment:
This is probably out of scope of this PR, but do you think it would be possible for us to call all these init functions with config as an argument instead of using global one explicitely? This would make it possible to unit test them to make sure that we don't mess things up, or is there anything aside from config manipulation that makes it unfeasible?

Reply:
Good idea! Added to my TODO list.

Reply:
Agree, would be nice to get some test coverage on this entire part of the code (i.e. dependent option handling)